### PR TITLE
EZP-31193: Added information about Site Access groups to the Site Access

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-EZP-31193](https://jira.ez.no/browse/EZP-31193)
| **Bug/Improvement**|no
| **New feature**    | yes
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes/no
| **Doc needed**     | yes/no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

**Comment by gggeek** [comment]
Shooting from the hip here, but:
although having this data seems nice, at the moment we have SF serializing the siteaccess in the URL for each request to an ESI controller.
Those urls are already quite fugly, and very long as well. Long urls give troubles to intermediate proxies, which need special config to be handle them by making their memory buffers larger than usual.
Was this PR tested in a context where varnish is in usage? Did the size of esi urls increase ?

File: eZ/Publish/Core/MVC/Symfony/SiteAccess/Provider/StaticSiteAccessProvider.php [hunk 1]
```diff
@@ -49,9 +56,22 @@ public function isDefined(string $name): bool
...
+
+    private function createSiteAccess(string $name): SiteAccess
+    {
+        $siteAccess = new SiteAccess($name, SiteAccess::DEFAULT_MATCHING_TYPE, null, self::class);
+        $siteAccess->groups = array_map(
+                static function ($groupName) {
+                    return new SiteAccessGroup($groupName);
+                },
+                $this->groupsBySiteAccess[$name]
+            ) ?? [];
```
Comment (adamwojs) [inline eZ/Publish/Core/MVC/Symfony/SiteAccess/Provider/StaticSiteAccessProvider.php]: `array_map` always returns an array so `?? []` is unnecessary or placed in the wrong place.

File: eZ/Publish/Core/MVC/Symfony/SiteAccess/Provider/StaticSiteAccessProvider.php [hunk 2]
```diff
@@ -49,9 +56,22 @@ public function isDefined(string $name): bool
...
     }
+
+    private function createSiteAccess(string $name): SiteAccess
+    {
+        $siteAccess = new SiteAccess($name, SiteAccess::DEFAULT_MATCHING_TYPE, null, self::class);
+        $siteAccess->groups = array_map(
+            static function ($groupName) {
+                return new SiteAccessGroup($groupName);
+            },
+            $this->groupsBySiteAccess[$name]
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/SiteAccess/Provider/StaticSiteAccessProvider.php]: is it possible to set up a SA w/o a group? (don't remember). If so, what would happen here in this case?

**Comment by mikadamczyk** [comment]
done, ping @alongosz @adamwojs

File: eZ/Publish/Core/MVC/Symfony/SiteAccess/Tests/Provider/ChainSiteAccessProviderTest.php [hunk 3]
```diff
@@ -10,33 +10,57 @@
...
         $this->providers = [
-            new StaticSiteAccessProvider([self::EXISTING_SA_NAME, 'first_sa']),
-            new StaticSiteAccessProvider(['second_sa']),
+            new StaticSiteAccessProvider([self::EXISTING_SA_NAME, 'first_sa'], $this->groupsBySiteAccess),
+            new StaticSiteAccessProvider(['second_sa'], $this->groupsBySiteAccess),
+            new StaticSiteAccessProvider([self::WITHOUT_GROUP_SA_NAME]),
+        ];
+    }
+
+    public function isDefinedProvider(): array
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/SiteAccess/Tests/Provider/ChainSiteAccessProviderTest.php]: Nitpick: `is...` usually indicates `: bool`, which may be misleading.

**Comment by alongosz** [comment]
> Shooting from the hip here, but:
> although having this data seems nice, at the moment we have SF serializing the siteaccess in the URL for each request to an ESI controller.
> Those urls are already quite fugly, and very long as well. Long urls give troubles to intermediate proxies (and to varnish itself), which need special config to be handle them by making their memory buffers larger than usual.
> Was this PR tested in a context where varnish is in usage? Did the size of esi urls increase ?

Our QA could take a look when testing it.

**Pull request merged**